### PR TITLE
:recycle: Content Development Hero Section #156

### LINKED ISSUE #156: 🎨 Create Content Development Hero Section
#  Create the content development hero section 
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a nice hero section on the content development page that captures my attention and introduces the page content. Reuse the text-content component created in #20, and the reusable slider button in #24. I want a good experience on the web and phone.

Design:

content development header

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
 
### Technical Analysis

This feature can be found in libs/features/pages/content-development inside a components folder. The component's name is content-dev-hero-section.

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the global style.scss
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by preston-56**
# Description
 - Created a hero section component on the content development page that introduces the page content.
 - The component reuses the text-content component created in https://github.com/italanta/elewa-website/issues/20, and the reusable slider button in https://github.com/italanta/elewa-website/issues/24.

Fixes #156 

## Screenshot
Screenshot from 2023-09-05 16-52-55

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  }
+/* Styling for individual feat-one, feat-two, and feat-three divs */
+.feat-one {
+    top: 737px;
+    left: 291px;
+    width: 462px;
+    height: 461px;
+    /**UI Properties*/
+    background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss:46]: Use CSS variables specified in elements/theming/_variables.scss, to specify colors for the linear gradient

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    margin-left: 1rem/* Adjust the margin value as needed */
+
+  }
+  
+  .feat-two {
+    top: 504px;
+    left: 830px;
+    width: 462px;
+    height: 800px; /* Adjust the height for feat-two */
+    background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss:60]: Same here

File: libs/features/pages/content-development/src/lib/features-pages-content-development.module.ts [hunk 3]
```diff
@@ -4,10 +4,23 @@ import { CommonModule } from '@angular/common';
 import { ContentDevelopmentPageComponent } from './main/content-development-page/content-development-page.component';
 
 import { ContentDevelopmentPageRoutingModule } from './content-development.routing';
+import { ContentDevHeroSectionComponent } from './content-dev-hero-section/content-dev-hero-section.component';
+import { TextsModule } from '@elewa-website/elements/layout/texts';
+import { AppHeaderModule } from '@elewa-website/elements/layout/header';
+import { ButtonsModule } from '@elewa-website/elements/layout/buttons';
```
Comment (AnnaL001) [inline libs/features/pages/content-development/src/lib/features-pages-content-development.module.ts:10]: Kindly, observe the order of imports; elements imports should appear before local imports

**Review by AnnaL001 — CHANGES_REQUESTED** [review]
Good job, just a few changes before this is merged.

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss (hunk 1 above)
Comment (preston-56) [reply libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss]: *resolved

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss (hunk 2 above)
Comment (preston-56) [reply libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    margin-left: 40px; /* Adjust the margin value as needed */
+
+  }
+  
+  .feat-three {
+    top: 276px;
+    left: 1369px;
+    width: 462px;
+    height: 1200px; /* Adjust the height for feat-three */
+    background: transparent linear-gradient(180deg, #F2CF73 0%, #F2CF7300 100%) 0% 0% no-repeat padding-box;
```
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+})
+export class ContentDevHeroSectionComponent {
+  heroTextData = {
+    title: 'Content Development',
+    description:
+      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor dolorem ipsum quia dolor sit Eos qui ratione voluptatem sequi nesciunt.',
+  };
+  sliderButtonData = {
+    text: 'Unlock your potential',
+  };
```
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/content-development/src/lib/content-dev-hero-section/content-dev-hero-section.component.ts]: @AnnaL001 *resolved

File: libs/features/pages/content-development/src/lib/features-pages-content-development.module.ts (hunk 3 above)
Comment (preston-56) [reply libs/features/pages/content-development/src/lib/features-pages-content-development.module.ts]: @AnnaL001 *resolved

**Comment by LemmyMwaura** [comment]
Very Good job on this but it's already been implemented.

**Pull request closed without merging**